base/src/main/java/org/aya/tyck/pat/PatTycker.java

```diff
@@ -374,9 +399,18 @@ private record PatData(
     tracing(TreeBuilder::reduce);
     addPatSubst(data.param.ref(), res, pat.sourcePos());
     data.results.append(res);
+
+    // update signature for current parameter,
+    // because it probably changed when we call `data.sig.inst`
+    exprTycker.localCtx.put(data.param().ref(), type);
```

Comment:
I suspect that the extra bindings come from here.

Reply:
of course

Reply:
Don't resolve, this is a bug.

Reply:
I'm taking care of it